packages/aws-cdk-lib/aws-iam/lib/role.ts

```diff
@@ -594,6 +594,10 @@ export class Role extends Resource implements IRole {
    * Grant permissions to the given principal to assume this role.
    */
   public grantAssumeRole(identity: IPrincipal) {
+    // Service Principals must use assumeRolePolicy
+    if (identity.policyFragment.principalJson.Service) {
+      throw new Error('Cannot use a Service Principal with grantAssumeRole, use assumeRolePolicy instead.');
+    }
```

Comment:
I think this would work?
```
    if (identity instanceof ServicePrincipal || identity instanceof AccountPrincipal) {
      throw new Error('Cannot use a service or account principal with grantAssumeRole, use assumeRolePolicy instead.');
    }
```

Reply:
Good call, this appears to have worked.  Added another test for account principal.